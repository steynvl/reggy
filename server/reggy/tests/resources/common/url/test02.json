{
  "type": "URL",
  "information": {
    "schemes": "file;ftp;http;https;ftp.;www.",
    "portNumbers": "Optional port number",
    "specOptionalPortNumbers": "",
    "specRequiredPortNumbers": "",
    "username": "No user names",
    "specUserNames": "",
    "password": "No password",
    "domainName": "Allow any domain name",
    "specDomainNames": "",
    "specificTld": "",
    "subdomainOnSpecDomain": "",
    "folders": "No folders",
    "minFolderDepth": "",
    "maxFolderDepth": "",
    "specFoldersOnly": "",
    "fileNames": "No file names",
    "specExtensions": "",
    "specFileNames": "",
    "optionalFileNames": false,
    "parameters": "No parameters",
    "specParameters": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
