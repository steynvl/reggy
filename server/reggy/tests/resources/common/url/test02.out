Java
Pattern regex = Pattern.compile("(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?");
(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?
Perl
my $regex = /(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?\/?/;
(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?\/?
POSIX
(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?/?
(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?/?
Python
regex = re.compile(r'(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?')
(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?
JavaScript
const regex = /(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?\/?/;
(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?\/?
PHP
$regex = '/(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?\/?/';
(?:(?:file|ftp|https?):\/\/|(?:ftp|www)\.)(?:[\da-z-]++\.)+[a-z]{2,63}+(?::\d{1,5}+)?\/?
Golang
regex, _ := regexp.Compile("(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]+\\.)+[a-z]{2,63}(?::\\d{1,5})?/?")
(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]+\\.)+[a-z]{2,63}(?::\\d{1,5})?/?
Rust
let re = Regex::new(r"(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?").unwrap();
(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?
C#
var re = new Regex(@"(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?");
(?:(?:file|ftp|https?)://|(?:ftp|www)\.)(?:[\da-z-]+\.)+[a-z]{2,63}(?::\d{1,5})?/?
Scala
val re = "(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?".r
(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?
Kotlin
val regex = Regex("(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?")
(?:(?:file|ftp|https?)://|(?:ftp|www)\\.)(?:[\\da-z-]++\\.)+[a-z]{2,63}+(?::\\d{1,5}+)?/?